app/scripts/controllers/newOverview.js

```diff
@@ -0,0 +1,1200 @@
+    var pipelineBuildConfigs = [];
+    overview.deploymentConfigsByPipeline = {};
+    state.pipelinesForDeploymentConfig = {};
+    _.each(overview.buildConfigs, function(buildConfig) {
+      if (!isJenkinsPipelineStrategy(buildConfig)) {
+        return;
+      }
+
+      pipelineBuildConfigs.push(buildConfig);
+
+      // TODO: Handle other types.
+      // Preserve the order they appear in the annotation.
+      var dcNames = BuildsService.usesDeploymentConfigs(buildConfig);
+      var bcName = getName(buildConfig);
+      _.set(overview, ['deploymentConfigsByPipeline', bcName], dcNames);
+      _.each(dcNames, function(dcName) {
+        state.pipelinesForDeploymentConfig[dcName] = state.pipelinesForDeploymentConfig[dcName] || [];
+        state.pipelinesForDeploymentConfig[dcName].push(buildConfig);
+      });
+    });
+
+    overview.pipelineBuildConfigs = _.sortBy(pipelineBuildConfigs, 'metadata.name');
+    updatePipelineOtherResources();
+    updatePipelineLabelSuggestions(overview.pipelineBuildConfigs);
+    updateFilterDisabledState();
+  };
+
+  // Find build configs with an output image that matches the deployment config
+  // image change trigger.
+  //
+  // Updates `state.buildConfigsByObjectUID`
+  //   key: deployment config UID
+  //   value: array of build configs, sorted by name
+  var matchOutputImagesToImageChangeTriggers = function() {
+    state.buildConfigsByObjectUID = {};
+    _.each(overview.deploymentConfigs, function(deploymentConfig) {
+      var buildConfigs = [];
+      var triggers = _.get(deploymentConfig, 'spec.triggers');
+      _.each(triggers, function(trigger) {
+        var from = _.get(trigger, 'imageChangeParams.from');
+        if (!from) {
+          return;
+        }
+
+        var ref = imageObjectRef(from, deploymentConfig.metadata.namespace);
+        var buildConfigsForRef = buildConfigsByOutputImage[ref];
+        if (!_.isEmpty(buildConfigsForRef)) {
+          buildConfigs = buildConfigs.concat(buildConfigsForRef);
+        }
+      });
+
+      buildConfigs = _.sortBy(buildConfigs, 'metadata.name');
+      setBuildConfigsForDeploymentConfig(buildConfigs, deploymentConfig);
+      updateRecentBuildsForDeploymentConfig(deploymentConfig);
+    });
+  };
+
+  // Find the build configs that relate to each deployment config. Pipeline
+  // build configs are grouped using an annotation. Other build configs are
+  // grouping using output images matched against a deployment config image
+  // change trigger.
+  var groupBuildConfigsByDeploymentConfig = function() {
+    groupPipelineBuildConfigsByDeploymentConfig();
+    matchOutputImagesToImageChangeTriggers();
+  };
+
+  var groupRecentBuildsByDeploymentConfig = function() {
+    _.each(overview.deploymentConfigs, updateRecentBuildsForDeploymentConfig);
+  };
+
+  var groupBuilds = function() {
+    if(!state.builds || !overview.buildConfigs) {
+      return;
+    }
+
+    // Reset these maps.
+    overview.recentPipelinesByBuildConfig = {};
```

Comment:
I'm hesitant to put things in `state` that aren't needed by list row. Then it's not obvious when you change something here if it impacts the list row component.